### PR TITLE
Finish typing, TypeScript checks enabled

### PR DESCRIPTION
**Pull request by askvortsov1**
Fix remaining type issues in our current subset of converted TS code, re-enable errors if typecheck GH action fails.

### CODE REVIEW

**Review by SychO9 — APPROVED** [review]

File: js/src/forum/components/ForgotPasswordModal.tsx [hunk 1]
```diff
@@ -1,34 +1,31 @@
 import app from '../../forum/app';
-import Modal from '../../common/components/Modal';
+import Modal, { IInternalModalAttrs } from '../../common/components/Modal';
 import Button from '../../common/components/Button';
 import extractText from '../../common/utils/extractText';
 import Stream from '../../common/utils/Stream';
+import Mithril from 'mithril';
+import RequestError from '../../common/utils/RequestError';
```
Comment (davwheat) [inline js/src/forum/components/ForgotPasswordModal.tsx:7]:
```suggestion
import type Mithril from 'mithril';
import type RequestError from '../../common/utils/RequestError';
```

File: js/src/forum/components/LogInModal.tsx [hunk 2]
```diff
@@ -1,44 +1,40 @@
...
+import Modal, { IInternalModalAttrs } from '../../common/components/Modal';
 import ForgotPasswordModal from './ForgotPasswordModal';
 import SignUpModal from './SignUpModal';
 import Button from '../../common/components/Button';
 import LogInButtons from './LogInButtons';
 import extractText from '../../common/utils/extractText';
 import ItemList from '../../common/utils/ItemList';
 import Stream from '../../common/utils/Stream';
+import type Mithril from 'mithril';
+import RequestError from '../../common/utils/RequestError';
```
Comment (davwheat) [inline js/src/forum/components/LogInModal.tsx:11]:
```suggestion
import type RequestError from '../../common/utils/RequestError';
```

**Review by davwheat — CHANGES_REQUESTED** [review]

File: js/src/common/components/EditUserModal.tsx [hunk 3]
```diff
@@ -239,9 +261,8 @@ export default class EditUserModal extends Modal {
 
   /**
    * @internal
-   * @protected
    */
-  userIsAdmin(user) {
-    return user.groups().some((g) => g.id() === Group.ADMINISTRATOR_ID);
+  protected userIsAdmin(user: User | null) {
+    return user && (user.groups() || []).some((g) => g?.id() === Group.ADMINISTRATOR_ID);
```
[the remark this answers is not in the document]
Comment (askvortsov1) [reply js/src/common/components/EditUserModal.tsx:266]: That's not really accurate, see [Playground Link](https://www.typescriptlang.org/play?#code/C4TwDgpgBAKlC8UDeAoK6oEMBcUBiArgHYDGwAlgPZEDcKAvlAD5TEAmEAZuURG3ShLUAzsCgQAHuVHDccRKgxZcACgCUCAHxQA5Jh0MUgkWKKVgAUSky5CVkQ7debLMKxEQRoUWGUANhAAdH6UAOYqktLAwgD8gZjqasY+-kEh4WaW1tFxCWpqQA)

It's not that `x.a` might or may not be defined, but that `x` itself might or might not be defined.

[the remark this answers is not in the document]
Comment (davwheat) [reply js/src/common/components/EditUserModal.tsx:266]: Gotcha!

File: js/src/admin/components/ReadmeModal.tsx [hunk 4]
```diff
@@ -1,11 +1,21 @@
 import app from '../../admin/app';
-import Modal from '../../common/components/Modal';
+import Modal, { IInternalModalAttrs } from '../../common/components/Modal';
 import LoadingIndicator from '../../common/components/LoadingIndicator';
 import Placeholder from '../../common/components/Placeholder';
 import ExtensionReadme from '../models/ExtensionReadme';
+import type Mithril from 'mithril';
+import { Extension } from '../AdminApplication';
```
Comment (davwheat) [inline js/src/admin/components/ReadmeModal.tsx:7]:
```suggestion
import type { Extension } from '../AdminApplication';
```

File: js/src/common/components/EditUserModal.tsx [hunk 5]
```diff
@@ -1,32 +1,44 @@
...
+import Modal, { IInternalModalAttrs } from './Modal';
 import Button from './Button';
 import GroupBadge from './GroupBadge';
 import Group from '../models/Group';
 import extractText from '../utils/extractText';
 import ItemList from '../utils/ItemList';
 import Stream from '../utils/Stream';
+import type Mithril from 'mithril';
+import type User from '../models/User';
+import { SaveAttributes, SaveRelationships } from '../Model';
```
Comment (davwheat) [inline js/src/common/components/EditUserModal.tsx:11]:
```suggestion
import type { SaveAttributes, SaveRelationships } from '../Model';
```

File: js/src/common/components/RequestErrorModal.tsx [hunk 6]
```diff
@@ -1,6 +1,12 @@
-import Modal from './Modal';
+import RequestError from '../utils/RequestError';
```
Comment (davwheat) [inline js/src/common/components/RequestErrorModal.tsx:1]:
```suggestion
import type RequestError from '../utils/RequestError';
```

File: js/src/common/states/ModalManagerState.ts [hunk 7]
```diff
@@ -1,5 +1,15 @@
+import Component from '../Component';
```
Comment (davwheat) [inline js/src/common/states/ModalManagerState.ts:1]:
```suggestion
import type Component from '../Component';
```

File: js/src/forum/components/ForgotPasswordModal.tsx (hunk 1 above)
Comment (davwheat) [reply js/src/forum/components/ForgotPasswordModal.tsx:7]: This one seems to have not been done.

File: js/src/forum/components/LogInModal.tsx (hunk 2 above)
Comment (davwheat) [reply js/src/forum/components/LogInModal.tsx:11]: Nor this

**Review by davwheat — APPROVED** [review]
Sorry, I missed some more type import suggestions last time.

After this, it's good to merge imo, but I'll probably be asleep when you do it.

**Pull request merged**